src/figuresClasses.ts

```diff
@@ -1,11 +1,103 @@
-export interface Figure {}
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  protected sidesArray: number[] = [this.a, this.b, this.c];
+
+  protected longestSide: number = this.sidesArray.splice(
+    this.sidesArray.indexOf(Math.max(...this.sidesArray)),
+  )[0];
+
+  protected otherSidesSum: number = this.sidesArray.reduce(
+    (acc: number, sideLength: number) => acc + sideLength,
+    0,
+  );
+
+  protected semiperimeter: number = (this.longestSide + this.otherSidesSum) / 2;
 
-export class Triangle implements Figure {}
+  protected isSidesWrong: boolean = this.longestSide >= this.otherSidesSum;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    readonly shape: Shape = Shape.triangle,
+  ) {
+    if (this.sidesArray.some((side) => side <= 0)) {
+      throw new Error('length of some side is <= 0');
+    }
 
-export class Rectangle implements Figure {}
+    if (this.isSidesWrong) {
+      throw new Error(
+        // eslint-disable-next-line max-len
+        'Sides`s values are wrong (the longest side of a triangle is >= than a sum of two others)',
+      );
+    }
+  }
+
+  getArea(): number {
+    return +Math.sqrt(
+      this.semiperimeter *
+        (this.semiperimeter - this.a) *
+        (this.semiperimeter - this.b) *
+        (this.semiperimeter - this.c),
+    ).toFixed(2);
```

Comment:
[CODE STYLE] - The `getArea` function doesn't need to accept any parameters. All necessary values are already available as object fields. Remove the parameters and use the instance fields directly.